nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -125,7 +125,7 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
       navigators_.push_back(class_loader_.createUniqueInstance(navigator_type));
       if (!navigators_.back()->on_configure(
           node, plugin_lib_names, feedback_utils,
-          &plugin_muxer_, odom_smoother_))
+          &plugin_muxer_, odom_smoother_, logger_cout_))
```

Comment:
Where is `logger_cout_` made? Shouldn't this be created in the BT Navigator to be passed onto the navigator base to share? I'm not sure why we're passing in a nullptr and its unused in the BT Navigator level unless (1) we're creating it here for all N instances to share or (2) we're keeping the logger object here to use somewhere.

(1) would mean we should be instantiating it above and then passing it in to share
(2) would mean we can remove modifying the `on_configure` API / storing the logger object at this high level, because its unused and breaks ABI (since its unused here anyway)

Reply:
yeah I know its weird to pass a nullptr but I got to this point to statisfy:

- There can only be one instance of [StdCoutLogger](https://github.com/BehaviorTree/BehaviorTree.CPP/blob/a574beaa39c25649ca42f5d529a6033e3d9f4327/src/loggers/bt_cout_logger.cpp#L12) so it has to be shared across all navigator plugins
- the logger can only be created with a `tree_` which is only available `BehaviorTreeNavigator` and not in `BTNavigator`

It's possible that I missed another obvious way of implementing though

Reply:
@facontidavide this is the second time that the singleton nature of the loggers has caused us problems in Nav2. We actually removed live Groot monitoring because of it. 

Would it be possible to allow multiple loggers in BT.CPP to allow for this?

Reply:
First, it is ok to remove the singleton constraint. Let me work on it.

Reply:
Done: https://github.com/BehaviorTree/BehaviorTree.CPP/commit/ced041bc3c00f450a83ee8bfc62471511bcc575d

Reply:
Note that creating your own logger is trivial and boring. StdCout is only 30 lines of code. Are you sure that you don't want to create your own, maybe one that use `RCLCPP_INFO` instead of `std::cout` ?

Reply:
I leave that to @tonynajjar , but I agree that doing an RCLCPP log would be better so it enters the `.ros/logs` entries. That would probably need to live here I guess since ROS shouldn't bleed into BT.CPP.

Reply:
> That would probably need to live here I guess since ROS shouldn't bleed into BT.CPP.

exactly!